router/internal/persistedoperation/operationstorage/redis/rdcloser.go

```diff
@@ -0,0 +1,72 @@
+package rd
+
+import (
+	"context"
+	"fmt"
+	"github.com/redis/go-redis/v9"
+	"go.uber.org/zap"
+	"strings"
+)
+
+type RDCloser interface {
+	redis.Cmdable
+	Close() error
+}
+
+type RedisCloserOptions struct {
+	Logger   *zap.Logger
+	URL      string
+	Password string
+}
+
+func NewRedisCloser(opts *RedisCloserOptions) (RDCloser, error) {
+	var rdb RDCloser
+	if !strings.Contains(opts.URL, ",") {
+		options, err := redis.ParseURL(opts.URL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse the redis connection url %s: %w", opts.URL, err)
+		}
+
+		if opts.Password != "" {
+			options.Password = opts.Password
+		}
+		rdb = redis.NewClient(options)
+	}
+
+	if !isFunctioningClient(rdb) || isClusterClient(rdb) {
+		opts.Logger.Info("Detected that redis is running in cluster mode.")
+		stripped := strings.ReplaceAll(opts.URL, "redis://", "")
+		rdb = redis.NewClusterClient(&redis.ClusterOptions{
+			Addrs:    strings.Split(stripped, ","),
```

Comment:
This can be done at the config level. We can split the value by comma through a struct tag.